addons/delivery/models/delivery_carrier.py

```diff
@@ -258,7 +258,8 @@ def fixed_rate_shipment(self, order):
                     'warning_message': False}
         price = self.fixed_price
         if self.company_id and self.company_id.currency_id.id != order.currency_id.id:
-            price = self.env['res.currency']._compute(self.company_id.currency_id, order.currency_id, price)
+            date = order.date_order or fields.Date.today()
+            price = self.env['res.currency']._convert(price, order.currency_id, self.company_id, date)
```

Comment:
@d-fence There was still a call to a deprecated method.
This tour is the only one going through checkout payment page when `website_sale_delivery` is installed.
I am not sure about `self.company_id` vs `order.company_id` as third parameter tho.

Reply:
As the condition is checking if self.company_id is truthy, I guess that we want to retrieve the conversion rate from the self.company_id. It seems logical to get the conversion rate from the company of the delivery method if set.

Reply:
Do you want to validate this commit by someone else? Otherwise you can r+ since I can't :yum: